src/main/mocha.ts

```diff
@@ -187,6 +187,10 @@ export default class MochaWrapper extends Mocha {
         forkArgs.push('--delay');
       }
 
+      if (this.options.grep) {
+        forkArgs.push('--grep');
+      }
```

Comment:
From what I can see this is the main issue: you pass `--grep` option but you don't pass the grep pattern. Probably you should use a code like this here:

```
if (this.options.grep) {
  forkArgs.push('--grep', this.options.grep.toString());
}
```

I use `toString` here because `mocha` stores grep pattern as a [regular expression](https://github.com/mochajs/mocha/blob/master/lib/mocha.js#L301): and toString() would allow you to safely get a representation of RegExp and pass it to subprocess as a string.